src/common/decorators/storage.ts

```diff
@@ -86,49 +88,70 @@ class Storage {
       }
     } catch (err: any) {
       // Safari in private mode doesn't allow localstorage
+    } finally {
+      if (this._listeners[storageKey]) {
+        this._listeners[storageKey].forEach((listener) =>
+          listener(oldValue, value)
+        );
+      }
     }
   }
 }
 
-const subscribeStorage = new Storage();
+const storages: Record<string, StorageClass> = {};
 
-export const LocalStorage =
-  (
-    storageKey?: string,
-    property?: boolean,
-    subscribe = true,
-    storageType?: globalThis.Storage,
-    propertyOptions?: PropertyDeclaration
-  ): any =>
+export const Storage =
```

Comment:
```suggestion
export const storage =
```

Should decorator be in lowercase?